internal/monitor.go

```diff
@@ -252,6 +252,49 @@ func scaleHpaTarget(ctx hpaScopedContext, targetKind string, namespace string, t
 	}
 }
 
+// Check behaviour section of HPA(if it exist) if scaling down is allowed by StabilizationWindowSeconds
+func allowedToScaleDown(ctx hpaScopedContext) (bool, error) {
+	// If replicas >0 we should be able to identify if scaling is allowed by behaviour from status.conditions
+	if ctx.hpa.Spec.Behavior != nil &&
+		ctx.hpa.Spec.Behavior.ScaleDown != nil &&
+		ctx.hpa.Spec.Behavior.ScaleDown.StabilizationWindowSeconds != nil {
+		for _, condition := range ctx.hpa.Status.Conditions {
+			if condition.Type == "AbleToScale" && condition.Status == v1.ConditionTrue && condition.Reason == "ReadyForNewScale" {
+				return true, nil
+			} else if (condition.Type == "AbleToScale" && condition.Status == v1.ConditionFalse) || (condition.Type == "AbleToScale" && condition.Reason == "ScaleUpStabilized") {
+				return false, nil
+			}
+		}
+	} else {
+		return true, nil
+	}
+	return false, fmt.Errorf("could not determine if scaling is allowed")
+}
```

Comment:
It took me some time to read through all this nesting to understand what's exactly going on. probably it's worth reorganizing it slightly different? 
```suggestion
// Check behaviour section of HPA(if it exist) if scaling down is allowed by StabilizationWindowSeconds
func allowedToScaleDown(ctx hpaScopedContext) (bool, error) {
	// If replicas >0 we should be able to identify if scaling is allowed by behaviour from status.conditions
	if ctx.hpa.Spec.Behavior == nil || ctx.hpa.Spec.Behavior.ScaleUp == nil || ctx.hpa.Spec.Behavior.ScaleDown.StabilizationWindowSeconds == nil {
		return true, nil
	}

	for _, condition := range ctx.hpa.Status.Conditions {
		if condition.Type == "AbleToScale" {
			if condition.Status == v1.ConditionTrue {
				return true, nil
			}
			if condition.Status == v1.ConditionFalse {
				return false, nil
			}

			return false, fmt.Errorf("could not determine if scaling is allowed, 'AbleToScale' condition status is '%s'", condition.Status)
		}
	}

	return false, fmt.Errorf("could not determine if scaling is allowed")
}
```